conda/core/index.py

```diff
@@ -158,24 +221,39 @@ def _make_virtual_package(name, version=None, build_string=None):
     )
 
 
-def _supplement_index_with_features(index, features=()):
+def _supplement_index_with_features(
+    index: dict[Any, Any], features: list[str] = []
```

Comment:
```suggestion
    index: dict[PackageRecord, PackageRecord], features: list[str] = []
```

It's safe to change this to `dict[PackageRecord, PackageRecord]`. The function `make_feature_record` always returns this type.